jmix-dynattr/dynattr-flowui/src/main/java/io/jmix/dynattrflowui/view/category/CategoryListView.java

```diff
@@ -0,0 +1,234 @@
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Objects;
+
+import static io.jmix.flowui.download.DownloadFormat.JSON;
+import static io.jmix.flowui.download.DownloadFormat.ZIP;
+
+@SuppressWarnings("SpringJavaInjectionPointsAutowiringInspection")
+@Route(value = "dynat/category", layout = DefaultMainViewParent.class)
+@ViewController("dynat_CategoryView.list")
+@ViewDescriptor("category-list-view.xml")
+@PrimaryListView(Category.class)
+@LookupComponent("categoriesGrid")
+@DialogMode(width = "47.5em")
+public class CategoryListView extends StandardListView<Category> {
+
+    private static final Logger log = LoggerFactory.getLogger(CategoryListView.class);
+
+    @Autowired
+    protected Notifications notifications;
+    @Autowired
+    protected Messages messages;
+    @Autowired
+    protected Metadata metadata;
+    @Autowired
+    protected DynAttrMetadata dynAttrMetadata;
+    @Autowired
+    protected EntityImportExport entityImportExport;
+    @Autowired
+    protected FetchPlans fetchPlans;
+    @Autowired
+    protected EntityImportPlans entityImportPlans;
+    @Autowired
+    protected Downloader downloader;
+    @Autowired
+    private AccessManager accessManager;
+
+
+    @ViewComponent
+    protected DataGrid<Category> categoriesGrid;
+    @ViewComponent
+    protected CollectionContainer<CategoryAttribute> attributesDc;
+    @ViewComponent
+    protected InstanceContainer<Category> categoryDc;
+    @ViewComponent
+    protected InstanceLoader<Category> categoryDl;
+    @ViewComponent
+    private CollectionLoader<Category> categoriesDl;
+    @ViewComponent
+    protected FileUploadField importField;
+    @ViewComponent
+    private Button applyChangesBtn;
+
+    @Subscribe
+    protected void onBeforeShow(BeforeShowEvent event) {
+        Objects.requireNonNull(attributesDc.getSorter()).sort(Sort.by(Sort.Direction.ASC, "orderNo"));
+        setupFieldsLock();
+    }
+
+    @Subscribe("categoriesGrid.applyChanges")
+    protected void onCategoriesGridApplyChanges(ActionPerformedEvent event) {
+        dynAttrMetadata.reload();
+        notifications.create(messages.getMessage(CategoryListView.class, "notification.changesApplied"))
+                .withType(Notifications.Type.SUCCESS)
+                .show();
+    }
+
+    @Install(to = "categoriesGrid.edit", subject = "afterSaveHandler")
+    private void categoriesGridEditAfterCommitHandler(Category category) {
+        categoriesDl.load();
+    }
+
+    @Install(to = "categoriesGrid.create", subject = "afterSaveHandler")
+    private void categoriesGridCreateAfterCommitHandler(Category category) {
+        categoriesDl.load();
+    }
+
+    @Subscribe(id = "categoriesDc", target = Target.DATA_CONTAINER)
+    protected void onCategoriesDcItemChange(InstanceContainer.ItemChangeEvent<Category> event) {
+        Category category = event.getItem();
```

Comment:
If I got it right, this Instance container is used to load related Attribute entities. Looks like WA. You can create collection container for Attribute entity and map it to `categoriesDc` declaratively using dataLoaderCoordinator